src/net.h

```diff
@@ -1126,6 +1127,20 @@ class CConnman
      */
     std::vector<CService> m_onion_binds;
 
+    /**
+     * Mutex protecting m_i2p_sam_sessions.
+     */
+    Mutex m_i2p_sessions_mutex;
+
+    /**
+     * A pool of created I2P SAM transient sessions that should be used instead
+     * of creating new ones in order to reduce the load on the I2P network.
+     * Creating a session in I2P is not cheap, thus if this is not empty, then
+     * pick an entry from it instead of creating a new session. If connecting to
+     * a host fails, then the created session is put to this pool for reuse.
+     */
+    std::queue<std::unique_ptr<i2p::sam::Session>> m_i2p_sessions GUARDED_BY(m_i2p_sessions_mutex);
```

Comment:
This would benefit from https://github.com/bitcoin/bitcoin/pull/25390 by just adding `Synced<...> m_i2p_sessions` without a separate mutex.